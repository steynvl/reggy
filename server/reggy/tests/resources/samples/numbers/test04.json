{
  "sampleStringsInfo": [
    [
      {
        "markerType": "Numbers",
        "markedStrings": [
          "adsads"
        ],
        "markerInfo": {
          "minValOfIntPart": 0,
          "maxValOfIntPart": 0,
          "decimalSeparator": "Comma",
          "minNrOfDecimals": 2,
          "maxNrOfDecimals": 7,
          "thousandSeparator": "None",
          "codePosition": "Before only",
          "currencySign": "None",
          "currencyCodes": "",
          "limitIntegerPart": false,
          "allowPlusSign": true,
          "allowMinusSign": true,
          "signIsRequired": false,
          "whitespaceAllowedAfterSign": false,
          "thousandSeparatorsAreRequired": false,
          "allowLeadingZeros": false,
          "requireIntegerPart": false,
          "allowExponent": false,
          "currencySignOrCodeRequired": false
        },
        "repeatInfo": {
          "repeat": "1"
        }
      }
    ]
  ],
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "samplesAndSemantics"
}
