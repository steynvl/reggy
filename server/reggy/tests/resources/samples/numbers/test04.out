Java
Pattern regex = Pattern.compile("[\\+-]\\+?\\d*(,\\d{2,7})?");
[\\+-]\\+?\\d*(,\\d{2,7})?
Perl
my $regex = /[\+-]\+?\d*(,\d{2,7})?/;
[\+-]\+?\d*(,\d{2,7})?
POSIX
[\+-]\+?\d*(,\d{2,7})?
[\+-]\+?\d*(,\d{2,7})?
Python
regex = re.compile(r'[\+-]\+?\d*(,\d{2,7})?')
[\+-]\+?\d*(,\d{2,7})?
JavaScript
const regex = /[\+-]\+?\d*(,\d{2,7})?/;
[\+-]\+?\d*(,\d{2,7})?
PHP
$regex = '/[\+-]\+?\d*(,\d{2,7})?/';
[\+-]\+?\d*(,\d{2,7})?
Golang
regex, _ := regexp.Compile("[\\+-]\\+?\\d*(,\\d{2,7})?")
[\\+-]\\+?\\d*(,\\d{2,7})?
Rust
let re = Regex::new(r"[\+-]\+?\d*(,\d{2,7})?").unwrap();
[\+-]\+?\d*(,\d{2,7})?
C#
var re = new Regex(@"[\+-]\+?\d*(,\d{2,7})?");
[\+-]\+?\d*(,\d{2,7})?
Scala
val re = "[\\+-]\\+?\\d*(,\\d{2,7})?".r
[\\+-]\\+?\\d*(,\\d{2,7})?
Kotlin
val regex = Regex("[\\+-]\\+?\\d*(,\\d{2,7})?")
[\\+-]\\+?\\d*(,\\d{2,7})?